bundles/org.openhab.binding.zm/src/main/java/org/openhab/binding/zm/internal/handler/ZmMonitorHandler.java

```diff
@@ -0,0 +1,293 @@
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.i18n.TimeZoneProvider;
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.types.RawType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerService;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.zm.action.ZmActions;
+import org.openhab.binding.zm.internal.config.ZmMonitorConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ZmMonitorHandler} represents a Zoneminder monitor. The monitor handler
+ * interacts with the server bridge to communicate with the Zoneminder server.
+ *
+ * @author Mark Hilbush - Initial contribution
+ */
+@NonNullByDefault
+public class ZmMonitorHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(ZmMonitorHandler.class);
+
+    private final TimeZoneProvider timeZoneProvider;
+
+    private @Nullable ZmBridgeHandler bridgeHandler;
+
+    private @NonNullByDefault({}) String monitorId;
+    private @Nullable Integer imageRefreshInterval;
+    private Integer alarmDuration = DEFAULT_ALARM_DURATION_SECONDS;
+
+    private @Nullable ScheduledFuture<?> imageRefreshJob;
+    private @Nullable ScheduledFuture<?> alarmOffJob;
+
+    public ZmMonitorHandler(Thing thing, TimeZoneProvider timeZoneProvider) {
+        super(thing);
+        this.timeZoneProvider = timeZoneProvider;
+    }
+
+    @Override
+    public void initialize() {
+        ZmMonitorConfig config = getConfigAs(ZmMonitorConfig.class);
+        monitorId = config.monitorId;
+        imageRefreshInterval = config.imageRefreshInterval;
+        Integer value = config.alarmDuration;
+        alarmDuration = value != null ? value : DEFAULT_ALARM_DURATION_SECONDS;
+        bridgeHandler = (ZmBridgeHandler) getBridge().getHandler();
+        updateStatus(ThingStatus.ONLINE);
+        startImageRefreshJob();
+    }
+
+    @Override
+    public void dispose() {
+        stopAlarmOffJob();
+        turnAlarmOff();
+        stopImageRefreshJob();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            // No need to handle as status will be updated at the next refresh interval
```

Comment:
Can you trigger an immediate refresh?

Reply:
In another binding I cached the channel states, then pulled from the cache when a refresh is received for the channel. I'll do the same thing here.

Reply:
Caching should only be done if an immediate poll is not feasible, e.g. the device sends only event-based and has no polling support. Calling the poll job would be the preferred solution here.

Reply:
Understood. But I don't want to poll for each channel that gets a REFRESH command. At startup, wouldn't that result in multiple polls of the server for each monitor, which clearly is not desirable. Do you have a suggestion for how to approach this in a way that won't cause multiple server polls?

Reply:
Many bindings face this challenge. For this use case [ExpiringCache ](https://www.openhab.org/javadoc/v2.5/org/eclipse/smarthome/core/cache/expiringcache) was introduced.

Reply:
When I read our conversation again, you might have implemented exactly this. This should be a cache with a few seconds expiring time.

Reply:
Thanks. I did something similar along the lines of what I did in the weathercompany binding. It was an easy change, and would require less rework than adapting to use ExpiringCache. I'll include it in what I push, then you can weigh in on whether it's acceptable.